diff --git a/include/jlcxx/module.hpp b/include/jlcxx/module.hpp
index fac0aee..79f2b79 100644
--- module.hpp
+++ module.hpp
@@ -886,12 +886,22 @@ struct ApplyType
 
 namespace detail
 {
-  template<typename T>
+ template<typename T>
   void finalize(T* to_delete)
   {
-    delete to_delete;
-  }
 
+    constexpr bool has_getImpl = requires(const T& t) {
+        t.getImpl();
+    };
+
+
+    if constexpr(has_getImpl) {
+        operator delete(to_delete);
+        // TODO? Will probably leak memory
+    } else {
+        delete to_delete;
+    }
+  }
   template<typename T>
   struct CreateParameterType
   {
diff --git a/include/jlcxx/stl.hpp b/include/jlcxx/stl.hpp
index 969a6a6..9d12636 100644
--- stl.hpp
+++ stl.hpp
@@ -79,7 +79,7 @@ void wrap_common(TypeWrapperT& wrapped)
   using T = typename WrappedT::value_type;
   wrapped.module().set_override_module(StlWrappers::instance().module());
   wrapped.method("cppsize", &WrappedT::size);
-  wrapped.method("resize", [] (WrappedT& v, const cxxint_t s) { v.resize(s); });
+ // wrapped.method("resize", [] (WrappedT& v, const cxxint_t s) { v.resize(s); });
   wrapped.method("append", [] (WrappedT& v, jlcxx::ArrayRef<T> arr)
   {
     const std::size_t addedlen = arr.size();
@@ -145,12 +145,12 @@ struct WrapValArray
   {
     using WrappedT = typename TypeWrapperT::type;
     using T = typename WrappedT::value_type;
-    wrapped.template constructor<std::size_t>();
+    //wrapped.template constructor<std::size_t>();
     wrapped.template constructor<const T&, std::size_t>();
     wrapped.template constructor<const T*, std::size_t>();
     wrapped.module().set_override_module(StlWrappers::instance().module());
     wrapped.method("cppsize", &WrappedT::size);
-    wrapped.method("resize", [] (WrappedT& v, const cxxint_t s) { v.resize(s); });
+//    wrapped.method("resize", [] (WrappedT& v, const cxxint_t s) { v.resize(s); });
     wrapped.method("cxxgetindex", [] (const WrappedT& v, cxxint_t i) -> const T& { return v[i-1]; });
     wrapped.method("cxxgetindex", [] (WrappedT& v, cxxint_t i) -> T& { return v[i-1]; });
     wrapped.method("cxxsetindex!", [] (WrappedT& v, const T& val, cxxint_t i) { v[i-1] = val; });
